plugins/menubar/macos/Classes/MenubarPlugin.m

```diff
@@ -20,10 +20,82 @@
 static NSString *const kMenuItemSelectedCallbackMethod = @"Menubar.SelectedCallback";
 static NSString *const kIdKey = @"id";
 static NSString *const kLabelKey = @"label";
+static NSString *const kShortcutKeyEquivalent = @"keyEquivalent";
+static NSString *const kShortcutSpecialKey = @"specialKey";
+static NSString *const kShortcutKeyModifiers = @"keyModifiers";
 static NSString *const kEnabledKey = @"enabled";
 static NSString *const kChildrenKey = @"children";
 static NSString *const kDividerKey = @"isDivider";
 
+static const int kFlutterShortcutModifierMeta = 1 << 0;
+static const int kFlutterShortcutModifierShift = 1 << 1;
+static const int kFlutterShortcutModifierAlt = 1 << 2;
+static const int kFlutterShortcutModifierControl = 1 << 3;
+
+// Returns the string used by NSMenuItem for the given special key.
+// See _shortcutSpecialKeyValues in menu_channel.dart for values.
+static NSString *KeyEquivalentCharacterForSpecialKey(int specialKey) {
+  unichar character;
+  switch (specialKey) {
+    case 1:
+      character = NSF1FunctionKey;
+      break;
+    case 2:
+      character = NSF2FunctionKey;
+      break;
+    case 3:
+      character = NSF3FunctionKey;
+      break;
+    case 4:
+      character = NSF4FunctionKey;
+      break;
+    case 5:
+      character = NSF5FunctionKey;
+      break;
+    case 6:
+      character = NSF6FunctionKey;
+      break;
+    case 7:
+      character = NSF7FunctionKey;
+      break;
+    case 8:
+      character = NSF8FunctionKey;
+      break;
+    case 9:
+      character = NSF9FunctionKey;
+      break;
+    case 10:
+      character = NSF10FunctionKey;
+      break;
+    case 11:
+      character = NSF11FunctionKey;
+      break;
+    case 12:
+      character = NSF12FunctionKey;
+      break;
+    case 13:
+      character = NSBackspaceCharacter;
+      break;
+    case 14:
+      character = NSDeleteCharacter;
+      break;
+    default:
+      return nil;
+  }
+  return [NSString stringWithCharacters:&character length:1];
+}
+
+// Returns the NSEventModifierFlags of |modifiers|, a value from kShortcutKeyModifiers.
+static NSEventModifierFlags KeyEquivalentModifierMaskForModifiers(NSNumber *modifiers) {
```

Comment:
Do we need to handle the other modifiers here? (NSEventModifierFlagCapsLock, NSEventModifierFlagNumericPad, NSEventModifierFlagHelp, NSEventModifierFlagFunction)

Reply:
This code only needs to handle the values that the platform channel can send, which is these four. If you mean do we need to add those on the Dart side too, same comment about actual use case applies to those too.

Reply:
My brief perusal of the docs suggested that numpad keys would be handled by a combination of number string and the NSEventModifierFlagNumericPad modifier, as opposed to the numpad keys having separate entries they way they are in LogicalKeyboardKey on the Dart side. It looks like this code can't distinguish between numpad number keys and regular numeric keys. Then there are similar questions about the other things that macOS considers modifiers which aren't in the set of allowed modifiers that this plugin defines.

But anyway, yeah, it's not clear how much people need that.